kernel/core/datamanagement/memory.cpp

```diff
@@ -3748,94 +3748,162 @@ std::vector<mu::value_type> Memory::getIndex(size_t col, const std::vector<mu::v
     return vIndex;
 }
 
-
 /////////////////////////////////////////////////
 /// \brief Calculates the simples form of a ANOVA
 /// F test
 ///
-/// \param colCategories size_t
+/// \param colCategories VectorIndex&
 /// \param colValues size_t
 /// \param _vIndex const VectorIndex&
 /// \param significance double
 /// \return AnovaResult
 ///
 /////////////////////////////////////////////////
-AnovaResult Memory::getOneWayAnova(size_t colCategories, size_t colValues, const VectorIndex& _vIndex, double significance) const
+std::vector<AnovaResult> Memory::getAnova(const VectorIndex& colCategories, size_t colValues, const VectorIndex& _vIndex, double significance) const
 {
     // Get indices
-    if (colCategories > memArray.size()
-        || !memArray[colCategories]
-        || memArray[colCategories]->m_type != TableColumn::TYPE_CATEGORICAL
+    if (colCategories[0] > memArray.size()
+        || !memArray[colCategories[0]]
+        || memArray[colCategories[0]]->m_type != TableColumn::TYPE_CATEGORICAL
         || significance >= 1.0
-        || significance <= 0.0)
+        || significance <= 0.0
+        || colCategories.size() > 2)
     {
         AnovaResult res;
         res.m_FRatio = NAN;
-        return res;
+        return std::vector<AnovaResult> {res};
     }
 
-    _vIndex.setOpenEndIndex(getElemsInColumn(colCategories)-1);
-    Memory _mem(2);
-    _mem.memArray[0].reset(memArray[colCategories]->copy(_vIndex));
-    _mem.memArray[1].reset(memArray[colValues]->copy(_vIndex));
+    bool isTwoWay = colCategories.size() == 2;
+    _vIndex.setOpenEndIndex(getElemsInColumn(colCategories[0])-1);
+
+    Memory _mem(colCategories.size()+1);
+    _mem.memArray[0].reset(memArray[colValues]->copy(_vIndex));
+    for(int i = 0; i < colCategories.size(); i++)
+        _mem.memArray[i+1].reset(memArray[colCategories[i]]->copy(_vIndex));
+
+    std::vector<std::vector<std::string>> factors;
+    for(int i = 0; i < colCategories.size(); i++)
+        factors.push_back(static_cast<CategoricalColumn*>(_mem.memArray[i+1].get())->getCategories());
 
-    const std::vector<std::string>& vCategories = static_cast<CategoricalColumn*>(_mem.memArray[0].get())->getCategories();
+    std::vector<Memory> groupedValues(std::pow(2, factors.size()) - 1);

```

Comment:
Either use `intPower(2, factors.size()) - 1` for that or `(1 << factors.size()) - 1`. `std::pow()` calls to `std::exp()` and `std::log()`, which is unprecise and slow for such an operation.